File: 7-Bit-Strings/sol.java
import java.util.*;
import java.lang.*;
import java.io.*;

class sol{
    // Comparator<> compare = new Comparator<>() {
    //     @Override
    //     public int compare( a, b) {
    //         return;
    //     }
    // };
    static long fast_power(long base, long power) {
        long MOD = 1000000007L;
        long result = 1;
        while(power > 0) {

            if(power % 2 == 1) { // Can also use (power & 1) to make code even faster
                result = (result*base) % MOD;
            }
            base = (base * base) % MOD;
            power = power / 2; // Can also use power >>= 1; to make code even faster
        }
        return result;
    }
    public static void  solve(){
        long n;
        Scanner sc=new Scanner(System.in);
        n = sc.nextLong();
        System.out.println(fast_power(2,n));
    }
    public static void main (String[] args) throws java.lang.Exception
	{
        Scanner sc=new Scanner(System.in);
		int t=1;
        // t = sc.nextInt();
        while(t>0){
            solve();
            t--;
        }
    }
}

File: 7-Bit-Strings/sol.cpp
#include <bits/stdc++.h>
using namespace std;

#define deb(x) cout << #x << "=" << x << endl
#define ll long long

#define MOD 1000000007
ll fast_power(ll base, ll power) {
    ll result = 1;
    while(power > 0) {

        if(power & 1) {
            result = (result*base) % MOD;
        }
        base = (base * base) % MOD;
        power = power >> 1; 
    }
    return result;
}
void solve()
{
	ll n;
	cin>>n; 
	cout<<fast_power(2,n)<<endl;
}


int main()
{
	ios_base::sync_with_stdio(0), cin.tie(0), cout.tie(0);
	int t = 1;
	// cin >> t;

	while (t--)
	{
		solve();
	}
	return 0;
}